tokio-postgres/src/query.rs

```diff
@@ -156,16 +156,18 @@ where
     I: IntoIterator<Item = P>,
     I::IntoIter: ExactSizeIterator,
 {
+    let param_types = statement.params();
     let (param_formats, params): (Vec<_>, Vec<_>) = params
         .into_iter()
-        .map(|p| (p.borrow_to_sql().encode_format() as i16, p))
+        .zip(param_types.iter())
```

Comment:
A zipped iterator stops as soon as the shortest iterator ends, so the length check below won't work properly. You may need to manually iterate.

Reply:
Would it be better (or sufficient) to move the length check before the `zip` to guard against the early-ending case?

Reply:
Oh yeah, we already require ExactSizeIterator so we can just move the assert.

Reply:
Great, updated [in this commit](https://github.com/sfackler/rust-postgres/pull/937/commits/4e0aaee8987fde55438d250130234e69023fe857).